sparta/sparta/utils/SpartaSharedPointer.hpp

```diff
@@ -525,11 +631,7 @@ namespace sparta
                 else {
                     if(SPARTA_EXPECT_FALSE(allocated_ > water_mark_)) {
                         if(!water_mark_warning_) {
```

Comment:
Do you have a sense of how probable it would be for modelers to use this watermark feature?
I was thinking if we could use ```SPARTA_EXPECT_TRUE``` for the if clause at line 633 if it is not likely.

Reply:
It's looking more and more likely as we're using it.  I'll add the macro.  Good point.

Reply:
Done.